### DISCUSSION TITLE
第4节课作业(1班)

### DISCUSSION THREAD

**Discussion by AXYZdong** [InternLM/Tutorial#268] [作业]
**基础作业：**

构建数据集，使用 XTuner 微调 InternLM-Chat-7B 模型, 让模型学习到它是你的智能小助手，效果如下图所示，本作业训练出来的模型的输出需要**将不要葱姜蒜大佬**替换成自己名字或昵称！

**微调前**（回答比较官方）
![web_show_2.png](https://github.com/InternLM/tutorial/blob/main/xtuner%2Fimgs%2Fweb_show_2.png)

**微调后**（对自己的身份有了清晰的认知）
![web_show_1.png](https://github.com/InternLM/tutorial/blob/main/xtuner%2Fimgs%2Fweb_show_1.png)

作业参考答案：https://github.com/InternLM/tutorial/blob/main/xtuner/self.md

**进阶作业：**

- 将训练好的Adapter模型权重上传到 OpenXLab、Hugging Face 或者 MoelScope 任一一平台。
- 将训练好后的模型应用部署到 OpenXLab 平台，参考部署文档请访问：https://aicarrier.feishu.cn/docx/MQH6dygcKolG37x0ekcc4oZhnCe

**Comment by woxinyonghen** [comment]
https://github.com/woxinyonghen/InternLMCamp/tree/main/homework03/基础作业/
https://github.com/woxinyonghen/InternLMCamp/tree/main/homework03/进阶作业/

**Comment by Gu-Xiao** [comment]
https://zhuanlan.zhihu.com/p/677366109

**Comment by biscuit279** [comment]
https://blog.csdn.net/qq_46212981/article/details/135556917

**Comment by xujinzh** [comment]
https://xujinzh.github.io/2024/01/12/ai-internlm-personal-assistant-based-on-xtuner/

**Comment by ustbzgn** [comment]
https://blog.csdn.net/qq_46052444/article/details/135567549?spm=1001.2014.3001.5501

**Comment by gpt-dance** [comment]
https://zhuanlan.zhihu.com/p/677614827

**Comment by zpc-666** [comment]
https://blog.csdn.net/zpc1219/article/details/135571472
作业截图见5.6小节

**Comment by strong-leaf** [comment]
https://zhuanlan.zhihu.com/p/677546293
作业见后

**Comment by UncleLLD** [comment]
https://openxlab.org.cn/apps/detail/unclell/personal-assistant-unclelld

![https://s11.ax1x.com/2024/01/13/pFPgKc6.png](https://s11.ax1x.com/2024/01/13/pFPgKc6.png)

**Comment by 10086zgyd** [comment]
https://github.com/10086zgyd/-XTuner-InternLM-Chat-7B-/blob/main/image.png

**Comment by Xpg74138** [comment]
https://github.com/Xpg74138/InternLMCamp/blob/main/Third_homework.ipynb

**Comment by AXYZdong** [comment]
https://nuly9zxzf1.feishu.cn/docx/KnFwdgJYfoHpiQxiSV7cN5hNnsb#DtiOdqpFXoLkGrxAcqTcgenxnk2

**Comment by grey2818** [comment]
https://zhuanlan.zhihu.com/p/677977573

**Comment by raulming** [comment]
https://www.notion.so/a2e4e69a582d46e49fd2f1ffe11f1887。见笔记末尾

**Comment by blurmemo** [comment]
https://github.com/blurmemo/LLM/blob/main/note/InternLM/homework/chap3-homework.md

**Comment by xiaozhuang223** [comment]
https://zhuanlan.zhihu.com/p/677611120
